src/lib/marlin_plonk_bindings/stubs/src/arkworks/dlog_proofs.rs

```diff
@@ -0,0 +1,125 @@
+use mina_curves::pasta::{pallas, vesta};
+use plonk_protocol_dlog::prover::ProverProof as DlogProof;
+use std::ops::Deref;
+
+// there are two curves we prove with
+
+type DlogProofPallas = DlogProof<pallas::Affine>;
+type DlogProofVesta = DlogProof<vesta::Affine>;
+
+// Pallas
+
+#[derive(Clone)]
+pub struct CamlDlogProofPallas(pub DlogProofPallas);
+
+unsafe impl ocaml::FromValue for CamlDlogProofPallas {
+    fn from_value(value: ocaml::Value) -> Self {
```

Comment:
This doesn't do what we want; we want to allocate an OCaml record as the return value for this, which matches the record type in `types.ml`.

Reply:
yeah you're right... This is what I don't like with this OCaml->Rust FFI, it is very error prone and not much is here to catch you. I was thinking that it'd be much better if the ocaml code was automatically generated from Rust. (For example, Rust has that for C: https://github.com/rust-lang/rust-bindgen)